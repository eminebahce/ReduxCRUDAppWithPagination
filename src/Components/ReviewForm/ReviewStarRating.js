import React from 'react'
import StarRatingComponent from 'react-star-rating-component';
import PropTypes from 'prop-types';


/*
    react-star-rating-component
    https://github.com/voronianski/react-star-rating-component
*/

class ReviewStarRating extends React.Component {

  onStarClick = (nextValue, prevValue) => {
    this.props.rate(nextValue);
  };

  render() {
    return (
        <StarRatingComponent
            name="star-rating"
            onStarClick={this.onStarClick}
            value={this.props.value}
            data-test="star-rating"
        />
    );
  }
}
ReviewStarRating.propTypes = {
  onStarClick : PropTypes.func
}

export default ReviewStarRating