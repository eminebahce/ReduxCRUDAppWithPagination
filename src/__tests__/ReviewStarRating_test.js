import React from 'react';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import ReviewStarRating from '../../src/Components/ReviewForm/ReviewStarRating';
import {Provider} from "react-redux";
import store from "../store";
import renderer from 'react-test-renderer';


describe('<ReviewStarRating /> component', () => {
    beforeEach(() => {
        const props = {
            onStarClick: () => {}
        }
    });
    const checkProps = (component, expectedProps) => {
        const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
        return propsErr;
    };
    it('should render without error', () => {
        const component = shallow(<ReviewStarRating />);
        const wrapper = component.find(`[data-test='star-rating']`);
        expect(wrapper.length).toBe(1)
    });

    describe('Checking propTypes', () => {
        it('should not throw a warning', () => {
            const expectedProps = {
                onStarClick: () => {}
            };
            const propsError = checkProps(ReviewStarRating, expectedProps);
            expect(propsError).toBeUndefined();
        });
    });

    describe('renders ReviewStarRating element', () => {
        it('should not return error', () => {
            const component = renderer.create(
                <ReviewStarRating />
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

});
