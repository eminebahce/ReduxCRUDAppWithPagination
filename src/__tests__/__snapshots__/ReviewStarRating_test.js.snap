// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewStarRating /> component renders ReviewStarRating element should not return error 1`] = `
<div
  className="dv-star-rating"
  style={
    Object {
      "display": "inline-block",
      "position": "relative",
    }
  }
>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="star-rating_5"
    name="star-rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={5}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="star-rating_5"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="star-rating_4"
    name="star-rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={4}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="star-rating_4"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="star-rating_3"
    name="star-rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={3}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="star-rating_3"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="star-rating_2"
    name="star-rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={2}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="star-rating_2"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="star-rating_1"
    name="star-rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={1}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="star-rating_1"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
</div>
`;
